staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -244,13 +244,15 @@ func (cnc *CloudNodeController) updateNodeAddress(ctx context.Context, node *v1.
 		}, nil
 	}
 
-	instanceMeta, err := instanceMetadataGetter(node.Spec.ProviderID, node)
-	if err != nil {
-		utilruntime.HandleError(err)
-		return
+	if instanceMetadata == nil {
+		var err error
+		if instanceMetadata, err = instanceMetadataGetter(node.Spec.ProviderID, node); err != nil {
```

Comment:
remove `instanceMetadataGetter` and call getInstanceMetadata directly?

Reply:
There is a difference between them: instanceMetadataGetter only get instance nodeAddresses, but getInstanceMetadata get both instance type and nodeAddresses.

Reply:
Maybe an option whether to get instance type make sense.